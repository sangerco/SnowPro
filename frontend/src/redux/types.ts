export interface UserData {
    username: string;
    firstName: string;
    lastName: string;
    email: string;
    avatar: string;
    bio: string;
    videos: string[];
    photos: string[];
    favMountains: string[];
  }


export const FETCH_USER_DATA_REQUEST = 'FETCH_USER_DATA_REQUEST';
export const FETCH_USER_DATA_SUCCESS = 'FETCH_USER_DATA_SUCCESS';
export const FETCH_USER_DATA_FAILURE = 'FETCH_USER_DATA_FAILURE';


export interface FetchUserDataRequestAction {
    type: typeof FETCH_USER_DATA_REQUEST
}

export interface FetchUserDataSuccessAction {
    type: typeof FETCH_USER_DATA_SUCCESS;
    payload: UserData;
}

export interface FetchUserDataFailureAction {
    type: typeof FETCH_USER_DATA_FAILURE;
    payload: string;
}

export type UserActionTypes = 
    FetchUserDataRequestAction
    | FetchUserDataSuccessAction
    | FetchUserDataFailureAction;