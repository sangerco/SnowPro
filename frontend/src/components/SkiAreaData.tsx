import React, { useEffect, useState } from "react";
import { Container, Header, Dimmer, Loader, Icon } from "semantic-ui-react";

interface SkiAreaDataResponse {
    data: {
        name: string;
        location: {
            latitude: number;
            longitude: number;
        };
        twitter: {
            user: string;
        }
    };
}

const SkiAreaData = () => {
    const [ name, setName ] = useState<string | null>(null);
    const [ lat, setLat ] = useState<number | null>(null);
    const [ long, setLong ] = useState<number | null>(null);
    const [ twitterUser, setTwitterUser ] = useState<string | null>(null)

    useEffect(() => {
        const fetchSkiAreaData =async () => {
            try {
                const response = await fetch('http://localhost:5000/resort/example');
                const responseData: SkiAreaDataResponse = await response.json();
                setName(responseData.data.name);
                setLat(responseData.data.location.latitude);
                setLong(responseData.data.location.longitude);
                setTwitterUser(responseData.data.twitter.user);
            } catch (e) {
                console.error(e)
            }
        };

        fetchSkiAreaData();
    }, []);

    if(name) {
        return (
            <Container text>
                <Header size="huge">{ name }</Header>
                <Container textAlign='right'>
                    <p style={{ color: "royalblue" }}>Twitter: { twitterUser }</p>
                </Container>
                <Container textAlign='right'>
                    <p><small><span style={{ color: "royalblue" }}><Icon name='map outline' /></span> { lat }, { long }</small></p>
                </Container>
            </Container>
        )
    } else {
        return (
            <div>
                <Dimmer active>
                    <Loader>Loading...</Loader>
                </Dimmer>
            </div>
        )
    }
};

export default SkiAreaData;
