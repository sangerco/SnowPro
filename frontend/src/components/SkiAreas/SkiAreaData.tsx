import React, { useEffect, useState } from "react";
import { Dimmer, Loader } from "semantic-ui-react";
import { URL } from "../../config";
import SkiArea from "./SkiArea";
import axios from "axios";
import { SkiResort, SkiResortDataResponse } from "../../interfaces/skiAreaInterfaces";

const SkiAreaData = () => {
    const [skiAreaResponse, setSkiAreaResponse] = useState<SkiResort[]>([]);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        const fetchSkiAreaData =async () => {
            try {
                const response: SkiResortDataResponse = await axios.get(`${URL}/ski-areas`);
                setSkiAreaResponse(response.data);
                setLoading(false);
            } catch (e) {
                console.error(e);
                setLoading(false);
            }
        };

        fetchSkiAreaData();
    }, []);

    if(skiAreaResponse === null) {
        return (
            <div>
                <Dimmer active>
                    <Loader>Loading...</Loader>
                </Dimmer>
            </div>
        )
    } else {
        return (
            <div>
                {skiAreaResponse.map((skiArea) => (
                    <SkiArea 
                        key={skiArea.slug}
                        slug={skiArea.slug}
                        name={skiArea.name}
                        country={skiArea.country}
                        region={skiArea.region}
                        lat={skiArea.location.latitude}
                        long={skiArea.location.longitude}
                        url={skiArea.url}
                    />
            ))}
            </div>
        );
    }
};

export default SkiAreaData;
