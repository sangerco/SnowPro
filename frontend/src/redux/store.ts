import { Action, compose } from "redux";
import { ThunkAction, configureStore } from "@reduxjs/toolkit";
import { logger } from "redux-logger";
import rootReducer from "./reducers/rootReducer";

const composeEnhancers =
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(logger, composeEnhancers),
});
