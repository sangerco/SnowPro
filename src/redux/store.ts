import { configureStore, combineReducers } from "@reduxjs/toolkit";
import userReducer from "./slices/userSlice";
import authReducer from "./slices/authSlice";
import messageReducer from "./slices/messageSlice";
import messageReplyReducer from "./slices/messageReplySlice";
import reviewReducer from "./slices/reviewSlice";
import reviewReplyReducer from "./slices/reviewReplySlice";
import tagReducer from "./slices/tagSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";

const persistConfig = {
  key: "root",
  storage,
  blacklist: ["messages", "messageReplies", "reviews", "reviewReplies"],
};

const rootReducer = combineReducers({
  users: userReducer,
  auth: authReducer,
  messages: messageReducer,
  messageReplies: messageReplyReducer,
  reviews: reviewReducer,
  reviewReplies: reviewReplyReducer,
  tags: tagReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const persistor = persistStore(store);
