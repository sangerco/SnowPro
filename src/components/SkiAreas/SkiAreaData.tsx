import React, { useEffect, useState } from "react";
import {
  Card,
  Dimmer,
  Divider,
  Grid,
  Header,
  Loader,
  Rating,
} from "semantic-ui-react";
import { fetchAllReviews } from "../../redux/slices/reviewSlice";
import { URL } from "../../utils/config";
import SkiArea from "./SkiArea";
import axios from "axios";
import {
  SkiResort,
  SkiResortDataResponse,
} from "../interfaces/skiAreaInterfaces";
import { useDispatch, useSelector } from "react-redux";
import { AppDispatch, RootState } from "../../redux/store";
import { Link } from "react-router-dom";

const SkiAreaData = () => {
  const dispatch = useDispatch<AppDispatch>();
  const [skiAreaResponse, setSkiAreaResponse] = useState<SkiResort[]>([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchSkiAreaData = async () => {
      try {
        const response: SkiResortDataResponse = await axios.get(
          `${URL}/ski-areas`
        );
        setSkiAreaResponse(response.data);
        setLoading(false);
      } catch (e) {
        console.error(e);
        setLoading(false);
      }
    };

    fetchSkiAreaData();
  }, []);

  useEffect(() => {
    dispatch(fetchAllReviews());
  }, [dispatch]);

  const reviewState = useSelector((state: RootState) => state.reviews);
  const reviews = reviewState.reviews;

  const formatDate = (date: Date) => {
    const newDate = new Date(date);
    const options: Intl.DateTimeFormatOptions = {
      year: "numeric",
      month: "long",
      day: "numeric",
    };
    return newDate.toLocaleDateString("en-US", options);
  };

  if (skiAreaResponse === null) {
    return (
      <div>
        <Dimmer active>
          <Loader>Loading...</Loader>
        </Dimmer>
      </div>
    );
  } else {
    return (
      <Grid>
        <Grid.Row>
          <Grid.Column width={4} textAlign="center">
            {reviewState.loading ? (
              <Dimmer active>
                <Loader>Loading...</Loader>
              </Dimmer>
            ) : reviewState.error ? (
              <Dimmer active>
                <Header as="h1">
                  Error! Review Data cannot be retrieved! {reviewState.error}
                </Header>
              </Dimmer>
            ) : reviews && reviews.length > 0 ? (
              <>
                {reviews.map((review) => (
                  <Card style={{ marginLeft: "10px" }}>
                    <Card.Content key={review.id} id="review-card">
                      <Card.Header>
                        <Link to={`/ski-areas/reviews/${review.id}`}>
                          {review.header}
                        </Link>{" "}
                        <Divider />
                        {review.skiAreaName}
                      </Card.Header>
                      <Card.Description>By {review.username}</Card.Description>
                      <Card.Meta>
                        <Rating
                          icon="star"
                          defaultRating={review.stars}
                          maxRating={5}
                          disabled
                        />
                      </Card.Meta>
                      {formatDate(review.createdAt)}
                    </Card.Content>
                  </Card>
                ))}
              </>
            ) : null}
          </Grid.Column>
          <Grid.Column width={8}>
            {skiAreaResponse.map((skiArea) => (
              <SkiArea
                key={skiArea.slug}
                slug={skiArea.slug}
                name={skiArea.name}
                country={skiArea.country}
                region={skiArea.region}
                lat={skiArea.location.latitude}
                long={skiArea.location.longitude}
                url={skiArea.url}
              />
            ))}
          </Grid.Column>
          <Grid.Column width={4}></Grid.Column>
        </Grid.Row>
      </Grid>
    );
  }
};

export default SkiAreaData;
