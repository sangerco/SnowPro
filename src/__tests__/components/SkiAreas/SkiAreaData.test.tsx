/* eslint-disable testing-library/no-node-access */
import React from "react";
import SkiAreaData from "../../../components/SkiAreas/SkiAreaData";
import { render, fireEvent, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../../redux/store";
import { BrowserRouter } from "react-router-dom";

describe("test Ski Area Data", () => {
  it("should render without crashing", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <SkiAreaData />
        </BrowserRouter>
      </Provider>
    );
  });
  it("should match snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <SkiAreaData />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
