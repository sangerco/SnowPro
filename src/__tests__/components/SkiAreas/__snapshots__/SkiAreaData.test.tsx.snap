// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Nav Bar should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui grid"
  >
    <div
      class="row"
    >
      <div
        class="center aligned four wide column"
      >
        <div
          class="ui active transition visible dimmer"
          style="display: flex;"
        >
          <div
            class="content"
          >
            <div
              class="ui text loader"
            >
              Loading...
            </div>
          </div>
        </div>
      </div>
      <div
        class="eight wide column"
      />
      <div
        class="four wide column"
      />
    </div>
  </div>
</DocumentFragment>
`;
